uxarray/grid/utils.py

```diff
@@ -189,123 +189,200 @@ def _newton_raphson_solver_for_gca_constLat(
+    if len(face_nodes) < 3:
+        # The return array has dimension len(mask) x 2 x 2 filled with Int Fill Value
+        return np.full((len(mask), 2, 2), INT_FILL_VALUE)
+
+    face_edges_connectivity = face_edges_connectivity = np.zeros(
+        (len(face_nodes), 2), dtype=int
+    )
+
+    # Initialize the first edge
+    face_edges_connectivity[0] = [face_nodes[0], face_nodes[1]]
 
-    # Ensure counter-clockwise order of edge nodes
-    # Start with the first two nodes
-    face_edges[0] = [face_nodes_ind[0], face_nodes_ind[1]]
+    # Do the vectorized check for counter-clockwise order of edge nodes
+    face_edges_connectivity[:, 0] = face_nodes
+    face_edges_connectivity[:, 1] = np.roll(face_nodes, -1)
 
-    for idx in range(1, len(face_edges)):
-        if face_edges[idx][0] != face_edges[idx - 1][1]:
-            # Swap the node index in this edge if not in counter-clockwise order
-            face_edges[idx] = face_edges[idx][::-1]
+    # Ensure the last edge connects back to the first node to complete the loop
+    face_edges_connectivity[-1] = [face_nodes[-1], face_nodes[0]]
 
     # Fetch coordinates for each node in the face edges
-    lonlat_coordinates = np.array(
-        [
-            [
-                [
-                    np.mod(np.deg2rad(node_lon[node]), 2 * np.pi),
-                    np.deg2rad(node_lat[node]),
-                ]
-                for node in edge
-            ]
-            for edge in face_edges
-        ]
-    )
+    nodes = face_edges_connectivity.flatten()
+    lonlat_coordinates = np.column_stack(
+        (
+            np.mod(np.deg2rad(node_lon_sliced[nodes]), 2 * np.pi),
+            np.deg2rad(node_lat_sliced[nodes]),
+        )
+    ).reshape(-1, 2, 2)
 
     return lonlat_coordinates
+
+
+def _get_lonlat_rad_face_edge_nodes(
+    face_nodes_sliced, node_lon_sliced, node_lat_sliced
+):
+    """Construct an array to hold the edge lat lon in radian connectivity for
+    multiple faces in a grid.
+
+    This function processes sliced portions of the total grid data. Users must prepare the sliced versions of the data according to their needs before using this function.
+
+    Parameters
+    ----------
+    face_nodes_sliced : list of np.ndarray
+        Each element is an array of shape (n_nodes_sliced,) corresponding to the sliced face's node indices.
+    node_lon_sliced : np.ndarray, shape (n_nodes_sliced,)
+        The values of Grid.node_lon for the sliced portion, where n_nodes_sliced is the total number of nodes in the sliced portion.
+    node_lat_sliced : np.ndarray, shape (n_nodes_sliced,)
+        The values of Grid.node_lat for the sliced portion.
+
+    Returns
+    -------
+    faces_lonlat_coordinates : np.ndarray
+        An array of shape (n_faces, n_edges, 2, 2) containing the latitude and longitude coordinates
+        in radians for the edges of each face. It might contain dummy values if the grid has holes.
+
```

Comment:
This array here is extremely large and impractical to create for anything but the smallest of meshes. 

For a 30km MPAS mesh (~655k faces and 2mil Edges), this leads to an array size of easily over a trillion doubles, which would take up thousands of GB in storage.

Reply:
Unless the second dimension should be `n_max_face_edges`

Reply:
the dimension is `n_max_face_edges`

Reply:
> @amberchen122 Can you also help me update the documentation, please? And if necessary, add an example usage. Thanks

Please mark resolve and I can give a quick review.

Reply:
@rajeeja 

There might be some larger underlying changes needed before another review (see the comments below).